File: tests/fixtures/codegen/expected/go/queries/events/geteventstates.go
package events

import (
	"context"
	public "example.com/pgsid-fixture/generated/go/schema/public"
)

const GetEventStatesSQL = "SELECT states FROM public.events WHERE id = $1;"

type GetEventStatesParams struct {
	Id *int64 `db:"id"`
}
type GetEventStatesRow struct {
	States []public.EventState `db:"states"`
}

func (q *Queries) GetEventStates(ctx context.Context, params GetEventStatesParams) (GetEventStatesRow, error) {
	var row GetEventStatesRow
	err := q.db.QueryRow(ctx, GetEventStatesSQL, params.Id).Scan(&row.States)
	return row, err
}

File: tests/fixtures/codegen/expected/typescript/runtime/queries/events/GetEventStates.ts
import type { GetEventStatesParams, GetEventStatesRow } from "../../../types/queries/events/GetEventStates.js";
export type { GetEventStatesParams, GetEventStatesRow } from "../../../types/queries/events/GetEventStates.js";
import type * as pgsid from "./pgsid/queryable.js";
export const getEventStatesSql = "SELECT states FROM public.events WHERE id = $1;";
export async function getEventStates(db: pgsid.Queryable, params: GetEventStatesParams): Promise<GetEventStatesRow | undefined> {
    const result = await db.query(getEventStatesSql, [params["id"]]);
    const row = result.rows[0];
    if (row === undefined)
        return undefined;
    return row as GetEventStatesRow;
}
